src/ring_theory/adjoin/basic.lean

```diff
@@ -102,6 +113,70 @@ lemma adjoint_prod_le (s : set A) (t : set B) :
   adjoin R (set.prod s t) ≤ (adjoin R s).prod (adjoin R t) :=
 adjoin_le $ set.prod_mono subset_adjoin subset_adjoin
 
+lemma adjoin_le_prod (s) (t) : adjoin R ((set.insert (1, 0) ((linear_map.inl R A B) '' s)) ∪
+  (set.insert (0, 1) ((linear_map.inr R A B) '' t))) ≤ (adjoin R s).prod (adjoin R t) :=
```

Comment:
This makes the statement much easier to read:
```suggestion
lemma adjoin_le_prod (s) (t) :
  adjoin R ((set.insert (1, 0) ((linear_map.inl R A B) '' s)) ∪
            (set.insert (0, 1) ((linear_map.inr R A B) '' t))) ≤ (adjoin R s).prod (adjoin R t) :=
```

Reply:
I also wonder whether this is a cleaner statement:
```suggestion
lemma adjoin_le_prod (s) (t) :
  adjoin R ((linear_map.inl R A B '' (s ∪ {1})) ∪
            (linear_map.inr R A B '' (t ∪ {1})))) ≤ (adjoin R s).prod (adjoin R t) :=
```

Reply:
Hmm, yes, this is slightly better... let me modify also the other lemmas.

Reply:
Done!